src/core/net/dns_headers.hpp

```diff
@@ -0,0 +1,607 @@
+        kRdFlagOffset = 0,                       ///< RD Flag offset.
+        kRdFlagMask   = 0x01 << kRdFlagOffset,   ///< RD Flag mask.
+
+        kRaFlagOffset = 7,                       ///< RA Flag offset.
+        kRaFlagMask   = 0x01 << kRaFlagOffset,   ///< RA Flag mask.
+        kRCodeOffset  = 0,                       ///< RCODE field offset.
+        kRCodeMask    = 0x0f << kRCodeOffset,    ///< RCODE field mask.
+    };
+
+};
+
+/**
+ * This structure represents an Resource record body format (RR).
+ *
+ */
+OT_TOOL_PACKED_BEGIN
+struct ResourceRecordPoD
+{
+    uint16_t mType;    ///< The type of the data in RDATA section.
+    uint16_t mClass;   ///< The class of the data in RDATA section.
+    uint32_t mTtl;     ///< Specifies the maximum time that the resource record may be cached.
+    uint16_t mLength;  ///< The length of RDATA section in bytes.
+} OT_TOOL_PACKED_END;
+
+/**
+ * This class implements Resource Record body format (RR).
+ *
+ */
+class ResourceRecord: private ResourceRecordPoD
+{
+public:
+    /**
+     * This method returns the type of the resource record.
+     *
+     * @returns The type of the resource record.
+     */
+    uint16_t GetType(void) const { return HostSwap16(mType); }
+
+    /**
+     * This method sets the type of the resource record.
+     *
+     * @param[in]  aType The type of the resource record.
+     *
+     */
+    void SetType(uint16_t aType) { mType = HostSwap16(aType); }
+
+    /**
+     * This method returns the class of the resource record.
+     *
+     * @returns The class of the resource record.
+     */
+    uint16_t GetClass(void) const { return HostSwap16(mClass); }
+
+    /**
+     * This method sets the class of the resource record.
+     *
+     * @param[in]  aClass The class of the resource record.
+     *
+     */
+    void SetClass(uint16_t aClass) { mClass = HostSwap16(aClass); }
+
+    /**
+     * This method returns the time to live field of the resource record.
+     *
+     * @returns The time to live field of the resource record.
+     */
+    uint32_t GetTtl(void) const { return HostSwap32(mTtl); }
+
+    /**
+     * This method sets the time to live field of the resource record.
+     *
+     * @param[in]  aTtl The time to live field of the resource record.
+     *
+     */
+    void SetTtl(uint32_t aTtl) { mTtl = HostSwap32(aTtl); }
+
+    /**
+     * This method returns the length of the resource record.
+     *
+     * @returns The length of the resource record.
+     */
+    uint16_t GetLength(void) const { return HostSwap16(mLength); }
+
+    /**
+     * This method sets the length of the resource record.
+     *
+     * @param[in]  aLength The length of the resource record.
+     *
+     */
+    void SetLength(uint16_t aLength) { mLength = HostSwap16(aLength); }
+
+};
+
+/**
+ * This class implements Resource Record body format (RR).
+ *
+ */
+OT_TOOL_PACKED_BEGIN
+class ResourceRecordAAAA: public ResourceRecord
```

Comment:
nit: I know it may look weird, but we typically camel case acronyms as well (e.g. Dhcp, Dns, Aaaa, etc.).